prometheus-collectors/class-user-stats-collector.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Automattic\VIP\Prometheus;
+
+use Automattic\VIP\Utils\Context;
+
+use Prometheus\Gauge;
+use Prometheus\RegistryInterface;
+use Two_Factor_Core;
+
+class User_Stats_Collector implements CollectorInterface {
+	private Gauge $users_gauge;
+
+	private string $blog_id;
+	private const MAX_USERS_FOR_COUNT     = 1_000_000;
+	private const MAX_USERS_FOR_2FA_STATS = 5_000;
+	private const METRIC_OPTION           = 'vip-prom-users';
+	public const TFA_STATUS_ENABLED       = 'enabled';
+	public const TFA_STATUS_DISABLED      = 'disabled';
+	public const TFA_STATUS_DEFAULT       = 'unknown';
+
+	public function initialize( RegistryInterface $registry ): void {
+		$this->blog_id = Plugin::get_instance()->get_site_label();
+
+		$this->users_gauge = $registry->getOrRegisterGauge(
+			'user',
+			'count',
+			'Number of users by role and 2FA status',
+			[ 'site_id', 'role', 'tfa_status' ]
```

Comment:
What's the site_id here? is it the network site or the env ID?

Reply:
Network site

Reply:
`get_site_label()` takes care of it.

Reply:
You mean call it something like `network_site_id`? That makes sense, though `site_id` is the label name used by a number of the other existing collectors so perhaps better to follow suit.

Reply:
> That makes sense, though site_id is the label name used by a number of the other existing collectors so perhaps better to follow suit.

for network sites?? Hmm.. I don't prefer that because it is easy to confuse with GOOP . @sjinks what do you feel?

Reply:
We should try to use the newer terminology wherever possible (i.e. `network_site_id`), including updating the existing the collectors.

Reply:
> I don't prefer that because it is easy to confuse with GOOP site IDs which are also used a lot on Prom

I think the GOOP site IDs are mostly limited to the namespace label in Prom? But I see what you mean - would definitely be less ambiguous. I'll update here and can open a PR to switch the others.